wgengine/magicsock/magicsock.go

```diff
@@ -682,7 +682,15 @@ func (c *Conn) updateNetInfo(ctx context.Context) (*netcheck.Report, error) {
 	ni.OSHasIPv6.Set(report.OSHasIPv6)
 	ni.WorkingUDP.Set(report.UDP)
 	ni.WorkingICMPv4.Set(report.ICMPv4)
-	ni.PreferredDERP = c.maybeSetNearestDERP(report)
```

Comment:
What's this change about? It looks like we're doing all of `maybeSetNearestDERP` except the bit about not doing it if we don't have a connection to control? Is that not needed any more? Should we update `maybeSetNearestDERP` instead? or I think we could remove that function, because the only remaining caller is in tests?